server/src/main/java/org/elasticsearch/env/Environment.java

```diff
@@ -179,7 +170,7 @@ public Settings settings() {
      * The data location.
      */
     public Path[] dataFiles() {
-        return dataFiles;
+        return new Path[] { dataFile };
```

Comment:
Could we (naively) inline this method right now, or in an imminent follow-up? That would leave us with a bunch of places where we do things like this:

```
        for (Path dataPath : new Path[]{environment.dataFile()}) {
```

I think this is preferable to using `environment.dataFiles()[0]` to get hold of the unique data path using the knowledge that it's always a singleton array. Unrolling loops like this is obviously correct (indeed IntelliJ will do it for you).

Reply:
Yeah, we discussed the approach here to not try to do MDP legacy code removal all at once, keeping each change relatively self-contained, and this is one we agreed to hold for a follow-up.

Reply:
Acked. An imminent follow-up, or do we expect to merge things like #72278 first? We call this method in <30 places, so it's a pretty small change that IMO makes the subsequent larger cleanups much neater.

Reply:
Yeah, after this change, although we didn’t discuss a dependency on #72278. You’re making a compelling case there should be. Thank you!

Reply:
I opened https://github.com/elastic/elasticsearch/pull/72327